### PR TITLE
fix: noop on UpdateState for invalid misbehaviour

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #3636 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved error handling in client update processes to prevent system crashes when encountering unexpected message types.
- **Tests**
	- Enhanced testing for client state updates to include more comprehensive checks and validations.
- **Documentation**
	- Updated the changelog to reflect the fix in the `light-clients/07-tendermint` module.

### CODE REVIEW

File: modules/light-clients/07-tendermint/update.go [hunk 1]
```diff
@@ -131,10 +131,12 @@ func (cs *ClientState) verifyHeader(
 // UpdateState must only be used to update within a single revision, thus header revision number and trusted height's revision
 // number must be the same. To update to a new revision, use a separate upgrade path
 // UpdateState will prune the oldest consensus state if it is expired.
+// If the provided clientMsg is not of type of Header then the handler will noop and empty slice is returned.
 func (cs ClientState) UpdateState(ctx sdk.Context, cdc codec.BinaryCodec, clientStore storetypes.KVStore, clientMsg exported.ClientMessage) []exported.Height {
 	header, ok := clientMsg.(*Header)
 	if !ok {
-		panic(fmt.Errorf("expected type %T, got %T", &Header{}, clientMsg))
+		// clientMsg is invalid Misbehaviour and handler should noop
```
Comment (colin-axner) [inline modules/light-clients/07-tendermint/update.go:138]:
```suggestion
		// clientMsg is invalid Misbehaviour, no update necessary
```

is this more clear?

**Review by colin-axner — APPROVED** [review]
ACK, thanks for the quick fix! Lets backport?

**Comment by damiannolan** [comment]
Added backport labels for v7.5 and v8.3

**Review by crodriguezvega — APPROVED** [review]
Thanks, @damiannolan!

**Pull request merged**